AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/src/CMCs/LocalCMC.dfy

```diff
@@ -12,6 +12,43 @@ module {:options "/functionSyntax:4" } LocalCMC {
   import Types = AwsCryptographyMaterialProvidersTypes
   import Seq
 
+  // This limits the visibility of the LocalCMC
+  // Doing this ensures that the smithy-dafny `axiom`
+  // added because the Types.ICryptographicMaterialsCache
+  // has the @aws.polymorph#mutableLocalState trait added in the smithy model
+  // can not leak and create unsoundness.
+  export
+    // This shows that LocalCMC extends Types.ICryptographicMaterialsCache
+    // as well as revealing the constructors spec.
+    reveals LocalCMC
+    provides
+      // These are dependent types that are needed
+      Types,
+      Wrappers,
+      DafnyLibraries,
+      CacheEntry,
+      UInt,
+
+      // These are the implementations of the trait
+      LocalCMC.GetCacheEntry',
+      LocalCMC.PutCacheEntry',
+      LocalCMC.DeleteCacheEntry',
+      LocalCMC.UpdateUsageMetadata',
+      LocalCMC.GetCacheEntryEnsuresPublicly,
+      LocalCMC.PutCacheEntryEnsuresPublicly,
+      LocalCMC.DeleteCacheEntryEnsuresPublicly,
+      LocalCMC.UpdateUsageMetadataEnsuresPublicly,
+      LocalCMC.ValidState,
+      LocalCMC.InternalValidState,
```

Comment:
While reviewing the H-Keyring changes,
I have been trying to figure out when to use `InternalValidState` vs `ValidState`.

I will keep trying to figure it out,
but maybe we should a comment?

Reply:
`ValidState` only.

`InternalValidState` is only for the internal class. The problem is the way I export things, I have to export it because it is part of the trait...